### DISCUSSION TITLE
GraphQL single query for all website content

### DISCUSSION THREAD

**Discussion by vladyslavvolkov** [bn-digital/home#16] [Coding]
https://github.com/bn-digital/zuluforest/commit/bb4a90af9f8671c79fc792ac08b0454e26ce4d8a

**Comment by TelezhenkoPV** [comment]
It doesn't mean, that we returned to the concept of large object called Website with whole content of app?
You show the possibility of creating custom memorized hook with search-by-slug?

**Reply by vladyslavvolkov** [reply]
One difference, that option to search single content block is still available. Also, if you follow keys pattern as in bndigital.co or in this repo (meaning sort of `home:hero` and `home:about` for same Home route), that gives you a possibility to query like


```tsx
const { pathname } = useLocation()
  return (
    <ContentComponent variables={{ where: { slug_contains: `${pathname}:` } }}>
      {({ data, loading }) => {
        const pageContent: ContentFragment[] | undefined = data?.content
        return (
          <Skeleton loading={loading}>
            {pageContent?.map(block => (
              <Typography.Title>{block?.title}</Typography.Title>
            ))}
          </Skeleton>
        )
      }}
    </ContentComponent>
  )
```

That should give an idea, why `Content` was not included as component inside `Website` - [query API](https://strapi.io/documentation/developer-docs/latest/development/plugins/graphql.html#query-api) is available only for Collection types
